dashboard/lib/pd/survey_pipeline/daily_survey_retriever.rb

```diff
@@ -1,62 +1,64 @@
+
+      # Collect survey questions; ignore forms that don't have submissions.
+      form_ids_with_submissions =
+        ws_submissions.pluck(:form_id).uniq | facilitator_submissions.pluck(:form_id).uniq
 
-      OUTPUT_KEYS.each do |key|
-        context[key] ||= []
-        context[key] += results[key]
-      end
+      questions = form_ids_with_submissions.empty? ? []
+        : Pd::SurveyQuestion.where(form_id: form_ids_with_submissions)
 
-      context
+      [questions, ws_submissions, facilitator_submissions]
     end
 
-    # Retrieve data from Pd::SurveyQuestion, Pd::WorkshopDailySurvey
-    # and Pd::WorkshopFacilitatorDailySurvey.
-    #
-    # @param filters [Hash{:workshop_ids, :form_ids => Array<Integer>}] contains array of selected
-    #   workshop ids form ids.
+    # Retrieve facilitator-specific survey submissions and questions for a facilitator
+    # in a selected list of workshops.
     #
-    # @return [Hash{:survey_questions, :workshop_submissions, :facilitator_submissions => Array}]
-    #   contains arrays of questions and submissions.
+    # @param facilitator_id [Integer]
+    # @param workshop_ids [Integer, Array<Integer>] a workshop id or a list of workshop ids
+    # @return [Array<Array>, nil] An array of 2 arrays: Array<Pd::SurveyQuestion> and
+    #   Array<Pd::WorkshopFacilitatorDailySurvey>.
+    #   Returns nil if either of the input parameter is nil.
     #
-    def self.retrieve_data(filters:)
-      workshop_ids = filters[:workshop_ids]
-      form_ids = filters[:form_ids]
+    def self.retrieve_facilitator_surveys(facilitator_id, workshop_ids)
+      return unless facilitator_id && workshop_ids
 
-      # Build where clause from filter values
-      submission_filter = {}
-      submission_filter[:pd_workshop_id] = workshop_ids if workshop_ids.present?
-      submission_filter[:form_id] = form_ids if form_ids.present?
+      submissions = Pd::WorkshopFacilitatorDailySurvey.where(
+        facilitator_id: facilitator_id, pd_workshop_id: workshop_ids
+      )
+      form_ids = submissions.pluck(:form_id).uniq
+      questions = Pd::SurveyQuestion.where(form_id: form_ids)
 
-      # Collect submissions
-      ws_submissions = Pd::WorkshopDailySurvey.where(submission_filter)
-      facilitator_submissions = Pd::WorkshopFacilitatorDailySurvey.where(submission_filter)
+      [questions, submissions]
+    end
 
-      # Collect survey questions; ignore forms that don't have submissions.
-      form_ids_with_submissions =
-        ws_submissions.pluck(:form_id).uniq | facilitator_submissions.pluck(:form_id).uniq
+    # Retrieve general workshop survey submissions and questions for selected workshops.
+    #
+    # @param workshop_ids [Integer, Array<Integer>] a workshop id or a list of workshop ids
+    # @return [Array<Array>] An array of 2 arrays: Array<Pd::SurveyQuestion> and
+    #   Array<Pd::WorkshopDailySurvey>. Returns nil if the input parameter is nil.
```

Comment:
Note: The `questions` and `submissions` inner return values from this function are not `Array`, they're [`ActiveRecord::Relation`](https://api.rubyonrails.org/classes/ActiveRecord/Relation.html).  If you don't want to expose that that level of detail to the consumer, you could just document them as `Enumerable`.

The same is true some other places in this file.

Reply:
Yeah that's more accurate. Will fix.